test/ng/compileSpec.js

```diff
@@ -12111,6 +12142,890 @@ describe('$compile', function() {
+        $rootScope.testUrl = $sce.trustAsResourceUrl('javascript:doTrustedStuff()');
+        $rootScope.$apply();
+
+        expect(element.prop('src')).toEqual('javascript:doTrustedStuff()');
+      }));
+    });
+
+    describe('base[href]', function() {
+      it('should be a RESOURCE_URL context', inject(function($compile, $rootScope, $sce) {
+        element = $compile('<base ng-prop-href="testUrl"/>')($rootScope);
+
+        $rootScope.testUrl = $sce.trustAsResourceUrl('https://example.com/');
+        $rootScope.$apply();
+        expect(element.prop('href')).toContain('https://example.com/');
+
+        $rootScope.testUrl = 'https://not.example.com/';
+        expect(function() { $rootScope.$apply(); }).toThrowMinErr(
+            '$sce', 'insecurl', 'Blocked loading resource from url not allowed by $sceDelegate policy.' +
+            '  URL: https://not.example.com/');
+      }));
+    });
+
+    describe('form[action]', function() {
+      it('should pass through action property for the same domain', inject(function($compile, $rootScope, $sce) {
+        element = $compile('<form ng-prop-action="testUrl"></form>')($rootScope);
+        $rootScope.testUrl = 'different_page';
+        $rootScope.$apply();
+        expect(element.prop('action')).toMatch(/\/different_page$/);
+      }));
+
+      it('should clear out action property for a different domain', inject(function($compile, $rootScope, $sce) {
+        element = $compile('<form ng-prop-action="testUrl"></form>')($rootScope);
+        $rootScope.testUrl = 'http://a.different.domain.example.com';
+        expect(function() { $rootScope.$apply(); }).toThrowMinErr(
+            '$sce', 'insecurl', 'Blocked loading resource from url not allowed by $sceDelegate policy.' +
+            '  URL: http://a.different.domain.example.com');
+      }));
+
+      it('should clear out JS action property', inject(function($compile, $rootScope, $sce) {
+        element = $compile('<form ng-prop-action="testUrl"></form>')($rootScope);
+        $rootScope.testUrl = 'javascript:alert(1);';
+        expect(function() { $rootScope.$apply(); }).toThrowMinErr(
+            '$sce', 'insecurl', 'Blocked loading resource from url not allowed by $sceDelegate policy.' +
+            '  URL: javascript:alert(1);');
+      }));
+
+      it('should clear out non-resource_url action property', inject(function($compile, $rootScope, $sce) {
+        element = $compile('<form ng-prop-action="testUrl"></form>')($rootScope);
+        $rootScope.testUrl = $sce.trustAsUrl('javascript:doTrustedStuff()');
+        expect(function() { $rootScope.$apply(); }).toThrowMinErr(
+            '$sce', 'insecurl', 'Blocked loading resource from url not allowed by $sceDelegate policy.' +
+            '  URL: javascript:doTrustedStuff()');
+      }));
+
+
+      it('should pass through $sce.trustAs() values in action property', inject(function($compile, $rootScope, $sce) {
```

Comment:
`trustAs()` --> `trustAsResourceUrl()`

Reply:
Again... make it different then the attr version?